libbeat/beat/beat.go

```diff
@@ -84,6 +85,7 @@ type BeatConfig struct {
 	Output  map[string]*ucfg.Config
 	Logging logp.Logging
 	Shipper publisher.ShipperConfig
+	Filter  []filter.FilterConfig
```

Comment:
I think `shipper` or `beat` (to be honest I think beat is too generic, maybe host?) should be the section that includes only the configuration options for that host like geoip configuration, max_queue, tags and shouldn't include `fields` and `filter` as these are not a configuration of the host and more of an enhancement of the output data. 
I suggest to leave it like this for now and change it in a separate PR if we decide t o structure differently the configuration file.